src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -52,7 +52,7 @@ public static void WaitWithoutInlining(this Task task)
             if (!task.IsCompleted)
             {
                 // Waiting on a continuation of a task won't ever inline the predecessor (in .NET 4.x anyway).
-                var continuation = task.ContinueWith(t => { }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+                var continuation = task.ContinueWith(t => { }, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
                 continuation.Wait();
```

Comment:
It does actually throw. Although we're missing a test to that effect, and I'll add one.

While this line doesn't throw, if you look below the if block, you'll see we rethrow any exception.